subsys/bluetooth/mesh/net.c

```diff
@@ -132,17 +132,30 @@ static bool msg_cache_match(struct bt_mesh_net_rx *rx,
 
 	for (i = 0U; i < ARRAY_SIZE(msg_cache); i++) {
 		if (msg_cache[i] == hash) {
+			rx->msg_cache_idx = i;
 			return true;
 		}
 	}
 
 	/* Add to the cache */
-	msg_cache[msg_cache_next++] = hash;
+	rx->msg_cache_idx = msg_cache_next++;
+	msg_cache[rx->msg_cache_idx] = hash;
 	msg_cache_next %= ARRAY_SIZE(msg_cache);
 
 	return false;
 }
 
+static void msg_cache_remove(struct bt_mesh_net_rx *rx,
+			     struct net_buf_simple *pdu)
+{
+	u64_t hash = msg_hash(rx, pdu);
+	if (msg_cache[rx->msg_cache_idx] == hash) {
+		msg_cache[rx->msg_cache_idx] = 0;
```

Comment:
This needs to be `0U` actually since it's a `u64_t` (see my other comment regarding MISRA-C)

Reply:
`u64_t` is defined as `unsigned long long`, so it should be `0ULL`. Ideally we should have something like `UINT64_C` in zephyr to simplify this.